examples/reshare/src/dkg/actor.rs

```diff
@@ -140,551 +151,493 @@ where
+                            response.acknowledge();
+                        }
                     }
-
-                    // Wait to acknowledge until the block has been processed by the application.
-                    //
-                    // If we did not block on processing the block, marshal could continue processing finalized blocks and start
-                    // at a future block after restart (leaving the application in an unrecoverable state where we are beyond the last epoch height
-                    // and not willing to enter the next epoch).
-                    response.acknowledge();
-                    info!(%epoch, relative_height, "finalized block");
                 }
             }
         }
-
-        // If the initial DKG was just performed, keep running until forcible exit.
-        if is_dkg {
-            // Close the mailbox to prevent accepting any new messages.
-            drop(self.mailbox);
-
-            // Exit last consensus instance to avoid useless work while we wait for shutdown (we
-            // won't need to finalize further blocks after the DKG completes).
-            orchestrator
-                .report(orchestrator::Message::Exit(current_epoch))
-                .await;
-
-            // Keep running until killed to keep the orchestrator mailbox alive, allowing
-            // peers that may have gone offline to catch up.
-            //
-            // The initial DKG process will never be exited automatically, assuming coordination
-            // between participants is manual.
-            info!("DKG complete...waiting for shutdown");
-            futures::future::pending::<()>().await;
-        }
-
-        info!("mailbox closed, exiting.");
+        info!("exiting DKG actor");
     }
 
-    fn select_participants(
-        current_epoch: Epoch,
-        num_participants: u32,
-        active_participants: Vec<C::PublicKey>,
-        inactive_participants: Vec<C::PublicKey>,
-    ) -> (Vec<C::PublicKey>, Vec<C::PublicKey>) {
-        let epoch0_players = Self::players_for_initial_epoch(
-            inactive_participants.clone(),
-            &active_participants,
-            num_participants,
-        );
-        if current_epoch.is_zero() {
-            return (active_participants, epoch0_players);
-        }
+    async fn handle_dealer_message<S: Sender<PublicKey = C::PublicKey>>(
+        storage: &mut Storage<ContextCell<E>, V, C::PublicKey>,
+        epoch: Epoch,
+        sender_pk: C::PublicKey,
+        pub_msg: DealerPubMsg<V>,
+        priv_msg: DealerPrivMsg,
+        player_state: &mut Player<V, C>,
+        network_sender: &mut S,
+    ) {
+        let response =
+            match player_state.handle(sender_pk.clone(), pub_msg.clone(), priv_msg.clone()) {
+                PlayerResponse::New(msg) => {
+                    storage
+                        .append_dealer_msg(epoch, sender_pk.clone(), pub_msg, priv_msg)
```

Comment:
why do we call this? Is handling to player not suffcient?

if not, this probably shouldn't live under a player-only call?